lib/IRGen/GenProto.cpp

```diff
@@ -722,6 +733,63 @@ bool EmitPolymorphicParameters::isClassPointerSource(unsigned paramIndex) {
   return false;
 }
 
+void EmitPolymorphicParameters::injectAdHocDistributedRequirements() {
+  // FIXME: We need a better way to recognize that function is
+  // a thunk for witness of `remoteCall` requirement.
+  if (!Fn.hasLocation())
```

Comment:
In my other PR I was actually going to add:

```
  /// Record which function this is a thunk for, we'll need this to link back
   /// calls in case this is a distributed requirement witness.
   thunk->getAttrs().add(
       new (C) DistributedThunkTargetAttr(func));
```

so we could check for it to detect a thunk; 
https://github.com/apple/swift/pull/70928/files#diff-c5ee3c123c92773a4f07b456c2c0051c7a9dfdc81c5c0234ba5da6389bd82a07R793

I'll cleanup  that PR shortly

Reply:
Great! Once that's there we can use it both here and in `FunctionSignatureOpts` optimizer pass.